pkg/client-sdk/types/types.go

```diff
@@ -96,6 +115,27 @@ func (t Transaction) String() string {
 	return string(buf)
 }
 
+type TxEventType int
+
+const (
+	TxsAdded TxEventType = iota
+	TxsSettled
+	TxsConfirmed
+)
+
+func (e TxEventType) String() string {
+	return map[TxEventType]string{
```

Comment:
shall we have also an even for preconf? or thats the default state of TxsAdded

Reply:
we don't handle preconf state yet